kyaml/fn/framework/frameworktestutil/frameworktestutil.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package frameworktestutil contains utilities for testing functions written using the framework.
+package frameworktestutil
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+// ResultsChecker tests a function by running it with predefined inputs and comparing
+// the outputs to expected results.
+type ResultsChecker struct {
+	// TestDataDirectory is the directory containing the testdata subdirectories.
+	// ResultsChecker will recurse into each test directory and run the Command
+	// if the directory contains both the ConfigInputFilename and at least one
+	// of ExpectedOutputFilname or ExpectedErrorFilename.
+	// Defaults to "testdata"
+	TestDataDirectory string
+
+	// ConfigInputFilename is the name of the config file provided as the first
+	// argument to the function.  Directories without this file will be skipped.
+	// Defaults to "config.yaml"
+	ConfigInputFilename string
+
+	// InputFilenameGlob matches function inputs
+	// Defaults to "input*.yaml"
+	InputFilenameGlob string
+
+	// ExpectedOutputFilname is the file with the expected output of the function
+	// Defaults to "expected.yaml".  Directories containing neither this file
+	// nore ExpectedErrorFilename will be skipped.
+	ExpectedOutputFilname string
+
+	// ExpectedErrorFilename is the file containing part of an expected error message
+	// Defaults to "error.yaml".  Directories containing neither this file
+	// nore ExpectedOutputFilname will be skipped.
+	ExpectedErrorFilename string
+
+	// Command provides the function to run.
+	Command func() cobra.Command
+}
+
+// Assert asserts the results for functions
+func (rc ResultsChecker) Assert(t *testing.T) bool {
+	if rc.TestDataDirectory == "" {
+		rc.TestDataDirectory = "testdata"
+	}
+	if rc.ConfigInputFilename == "" {
+		rc.ConfigInputFilename = "config.yaml"
+	}
+	if rc.ExpectedOutputFilname == "" {
+		rc.ExpectedOutputFilname = "expected.yaml"
+	}
+	if rc.ExpectedErrorFilename == "" {
+		rc.ExpectedErrorFilename = "error.yaml"
+	}
+	if rc.InputFilenameGlob == "" {
+		rc.InputFilenameGlob = "input*.yaml"
+	}
+
+	_ = filepath.Walk(rc.TestDataDirectory, func(path string, info os.FileInfo, err error) error {
```

Comment:
it's not too late to parameterize this framework with a file system (rather than committing to `filepath` and `os`).
In-mem tests have no container mounts, cleanup, os trouble.

There's api/filesys, which has no dependence on kustomize and supports [Walk](https://github.com/kubernetes-sigs/kustomize/blob/master/api/filesys/fsnode.go#L419). 
There's also https://github.com/spf13/afero
i wish Go had provided one out of the box; there's probably many of these now.

Reply:
Good to know.  Since this is a test framework itself, I suspect there is less value in being able to fake the fs -- we will be reading files checked into git.